waddrmgr/scoped_manager.go

```diff
@@ -1993,6 +1987,31 @@ func (s *ScopedKeyManager) importPublicKey(ns walletdb.ReadWriteBucket,
 	serializedPubKey, encryptedPrivKey []byte, addrType AddressType,
 	bs *BlockStamp) error {
 
+	// The start block needs to be updated when the newly imported address
+	// is before the current one.
+	startBlockModified := false
+	updateStartBlock := false
+	s.rootManager.mtx.Lock()
+	if bs.Height < s.rootManager.syncState.startBlock.Height {
+		updateStartBlock = true
+	}
+	s.rootManager.mtx.Unlock()
+
+	// Ensure the rootManager lock is taken outside the scoped manager lock, to
+	// avoid deadlocks.
+	defer func() {
```

Comment:
In the original code this code didn't run if there was an error before but now it will always run.

Reply:
I introduced the boolean `startBlockModified`, which is only set if the database update was made successfully, which is the place where the rootManager was updated before.

Reply:
I missed that. thanks.